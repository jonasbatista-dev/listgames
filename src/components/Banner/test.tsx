import { screen } from '@testing-library/react';

import Banner from '.';
import { renderWithTheme } from '../../utils/test/helpers';

const props = {
  img: 'https://source.unsplash.com/user/willianjusten/1042x580',
  title: 'Defy death',
  subtitle: '<p>Play the new <strong>CrashLands</strong> season</p>',
  buttonLabel: 'Buy now',
  buttonLink: '/games/defy-death',
};

describe('<Banner/>', () => {
  it('Should render the currectly', () => {
    const { container } = renderWithTheme(<Banner {...props} />);
    expect(container.firstChild).toMatchSnapshot();
    //title
    expect(
      screen.getByRole('heading', { name: /Defy death/i }),
    ).toBeInTheDocument();
    //subtitle
    expect(
      screen.getByRole('heading', {
        name: 'Play the new CrashLands season',
      }),
    ).toBeInTheDocument();
    //imagem
    expect(
      screen.getByRole('img', { name: /Defy death/i }),
    ).toBeInTheDocument();
  });

  it('Should render a Ribbon', () => {
    renderWithTheme(
      <Banner
        {...props}
        ribbon="Best Seller"
        ribbonSize="small"
        ribbonColor="secondary"
      />,
    );
    const ribbon = screen.getByText(/Best Seller/i);
    expect(ribbon).toHaveStyle({ 'background-color': '#3CD3C1' });
    expect(ribbon).toHaveStyle({ height: '2.6rem', 'font-size': '1.2rem' });
  });
});
