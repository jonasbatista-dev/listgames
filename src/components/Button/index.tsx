import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react';
import { Wrapper } from './styles';

type ButtonTypes =
  | AnchorHTMLAttributes<HTMLAnchorElement>
  | ButtonHTMLAttributes<HTMLAnchorElement>;

export type ButtonProps = {
  size?: 'large' | 'medium' | 'small';
  fullWidth?: boolean;
  icon?: React.ReactElement;
  as?: React.ElementType;
} & ButtonTypes;

const Button = ({
  children,
  size = 'medium',
  fullWidth,
  icon,
  ...props
}: ButtonProps) => {
  return (
    <Wrapper size={size} fullWidth={fullWidth} hasIcon={!!icon} {...props}>
      {!!icon && icon}
      {!!children && <span>{children}</span>}
    </Wrapper>
  );
};

export default Button;
