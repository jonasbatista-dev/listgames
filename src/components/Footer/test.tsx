import { render, screen } from '@testing-library/react';

import Footer from '.';

describe('<Footer/>', () => {
  it('Should render the heading', () => {
    const { container } = render(<Footer />);

    expect(
      screen.getByRole('heading', {
        name: /Footer/i,
      }),
    );

    expect(container?.firstChild).toMatchSnapshot();
  });
});
