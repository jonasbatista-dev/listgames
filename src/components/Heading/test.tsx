import { render, screen } from '@testing-library/react';

import Heading from '.';

describe('<Heading/>', () => {
  it('Should render the heading', () => {
    const { container } = render(<Heading />);

    expect(
      screen.getByRole('heading', {
        name: /Heading/i,
      }),
    );

    expect(container?.firstChild).toMatchSnapshot();
  });
});
