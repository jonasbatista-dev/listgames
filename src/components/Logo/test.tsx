import { screen } from '@testing-library/react';
import 'jest-styled-components';
import Logo from '.';
import { renderWithTheme } from '../../utils/test/helpers';

describe('<Logo/>', () => {
  it('Should render a white label by default', () => {
    renderWithTheme(<Logo />);

    expect(screen.getByLabelText(/LOGO/i).parentElement).toHaveStyle({
      color: '#FaFaFa',
    });
  });
  it('Should render a black label when color is passed', () => {
    renderWithTheme(<Logo color="black" />);

    expect(screen.getByLabelText(/LOGO/i).parentElement).toHaveStyle({
      color: '#030517',
    });
  });

  it('Should render a normal logo', () => {
    renderWithTheme(<Logo />);

    expect(screen.getByLabelText(/LOGO/i).parentElement).toHaveStyle({
      width: '11rem',
    });
  });
  it('Should render a bigger logo', () => {
    renderWithTheme(<Logo size="large" />);

    expect(screen.getByLabelText(/LOGO/i).parentElement).toHaveStyle({
      width: '20rem',
    });
  });

  // it('Should render a bigger logo without text if hideOnMobile', () => {
  //   renderWithTheme(<Logo hideOnMobile={true} />);

  //   expect(screen.getByLabelText(/LOGO/i).parentElement).toHaveStyleRule(
  //     'width',
  //     '5.8rem',
  //     {
  //       media: '(max-width: 768px)',
  //     },
  //   );
  // });
});
