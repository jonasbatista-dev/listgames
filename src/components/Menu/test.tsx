import { screen } from '@testing-library/react';

import Menu from '.';
import { renderWithTheme } from '../../utils/test/helpers';

describe('<Menu/>', () => {
  it('Should render the menu', () => {
    renderWithTheme(<Menu />);

    expect(screen.getByLabelText(/open menu/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/search/i)).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /won games/i })).toBeInTheDocument();
    expect(screen.getByLabelText(/shopping cart/i)).toBeInTheDocument();
  });
});
