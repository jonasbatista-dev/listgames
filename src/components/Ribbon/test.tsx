import { screen } from '@testing-library/react';

import Ribbon from '.';
import { renderWithTheme } from '../../utils/test/helpers';

describe('<Ribbon/>', () => {
  it('Should render the text correctly', () => {
    renderWithTheme(<Ribbon>Best Seller</Ribbon>);

    expect(screen?.getByText(/Best Seller/i)).toBeInTheDocument();
  });
  it('Should render with the primary color', () => {
    renderWithTheme(<Ribbon color="primary">Best Seller</Ribbon>);

    expect(screen?.getByText(/Best Seller/i)).toHaveStyle(
      'background-color: #F231A5',
    );
  });
  it('Should render with the normal size as default', () => {
    renderWithTheme(<Ribbon color="primary">Best Seller</Ribbon>);

    expect(screen?.getByText(/Best Seller/i)).toHaveStyle({
      height: '3.6rem',
      fontSize: '1.4rem',
    });
  });
  it('Should render with the small size', () => {
    renderWithTheme(<Ribbon size="small">Best Seller</Ribbon>);

    expect(screen?.getByText(/Best Seller/i)).toHaveStyle({
      height: '2.6rem',
      fontSize: '1.2rem',
    });
  });
});
