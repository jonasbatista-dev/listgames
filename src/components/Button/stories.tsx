import { Meta, StoryObj } from '@storybook/react';
import { MdAddShoppingCart } from 'react-icons/md';
import Button, { ButtonProps } from '.';

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    children: {
      type: 'string',
    },
    icon: {
      control: false,
    },
  },
} as Meta;

export const Default: StoryObj<ButtonProps> = {
  args: {
    children: 'Buy now',
  },
};

export const withIcon: StoryObj<ButtonProps> = {
  args: {
    size: 'small',
    children: 'Buy now',
    icon: <MdAddShoppingCart />,
  },
};
