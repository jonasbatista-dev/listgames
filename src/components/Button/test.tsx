import { screen } from '@testing-library/react';
import { MdAddShoppingCart } from 'react-icons/md';
import Button from '.';
import { renderWithTheme } from '../../utils/test/helpers';

describe('<Button/>', () => {
  it('Should render the medium size by default', () => {
    const { container } = renderWithTheme(<Button>Buy Now</Button>);
    // const { container } = render(<Button />);

    expect(
      screen.getByRole('button', {
        name: /Buy Now/i,
      }),
    ).toHaveStyle({
      padding: '0.8rem 3.2rem',
      height: '4rem',
      'font-size': '1.4rem',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should render the small size', () => {
    renderWithTheme(<Button size="small">Buy Now</Button>);
    // const { container } = render(<Button />);

    expect(
      screen.getByRole('button', {
        name: /Buy Now/i,
      }),
    ).toHaveStyle({
      padding: '0.8rem',
      height: '3rem',
      'font-size': '1.2rem',
    });
  });
  it('Should render the large size.', () => {
    renderWithTheme(<Button size="large">Buy Now</Button>);
    // const { container } = render(<Button />);

    expect(
      screen.getByRole('button', {
        name: /Buy Now/i,
      }),
    ).toHaveStyle({
      padding: '0.8rem 4.8rem',
      height: '5rem',
      'font-size': '1.6rem',
    });
  });
  it('Should render the fullwidth size', () => {
    renderWithTheme(<Button fullWidth>Buy Now</Button>);
    // const { container } = render(<Button />);

    expect(
      screen.getByRole('button', {
        name: /Buy Now/i,
      }),
    ).toHaveStyle({
      width: '100%',
    });
  });

  it('Should render an icon version', () => {
    renderWithTheme(
      <Button icon={<MdAddShoppingCart data-testid="icon" />}>Buy Now</Button>,
    );

    expect(screen.getByText(/Buy now/i)).toBeInTheDocument();
    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });
});
