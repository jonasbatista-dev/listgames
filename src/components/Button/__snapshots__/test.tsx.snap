// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Should render the medium size by default 1`] = `
.c0 {
  background: linear-gradient(180deg, #ff5f5f 0%, #f062f0 50%);
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  transition: opacity 0.2s ease;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c0:hover {
  background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
}

.c0:active {
  opacity: 0.5;
}

<button
  class="c0"
>
  <span>
    Buy Now
  </span>
</button>
`;
